packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -1967,17 +1968,36 @@ export default class Meeting extends StatelessWebexPlugin {
           this.logger.error('Meeting:index#join --> Error joining the call on roap initialization, ', err);
           throw err;
         })
-        .then(() =>
-          this.roap
-            .sendRoapMediaRequest({
-              sdp: this.mediaProperties.peerConnection.sdp,
-              roapSeq: this.roapSeq,
-              meeting: this // or can pass meeting ID
-            })
-            .catch((err) => {
-              this.logger.error('Meeting:index#join --> Error joining the call on send roap media request, ', err);
-              throw err;
-            }))
+        .then(() => new Promise((resolve, reject) => {
+          let timerCount = 0;
+          // eslint-disable-next-line func-names
+          // eslint-disable-next-line prefer-arrow-callback
+          const joiningTimer = setInterval(function () {
+            timerCount += 1;
+            if (this.meetingState === FULL_STATE.ACTIVE) {
+              clearInterval(joiningTimer);
+              resolve();
+            }
+
+            if (timerCount === 4) {
+              clearInterval(joiningTimer);
+              reject(new Error('Meeting is still not active '));
+            }
+          }.bind(this), 1000);
```

Comment:
This effectively just waits 4 seconds for the meeting to be Active? I guess for our specific case we know we need to wait for the device to join but for others it may be different?

Reply:
In 98% of the case the other user is JOINED with in a second so there wont be a wait. 
After /call api we get a mercury event which tells the meeting state to be JOINED. And then the /media request

We can fix this in two ways 
1) Set some flag on the addMedia wait for the mercury state change than do the api call.
     needs more work on store , action and if for some reason mercury event doesn't come the addMedia wont get triggered and user wont know it.
2) The above approach, I felt it was simple assuming the join event will surely come with in 4 seconds and if not we throw an promise error so the user can handle it.

Lets check with adam once and see what he thinks about it

Reply:
Logically /call /media and /mediaShare needs to be 3 separate calls but internally we do all three together